packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -124,8 +242,13 @@ const TimePicker = React.forwardRef(function TimePicker(
     };
 
     const mappedChildren = React.Children.map(children, (pickerSelect) => {
-      return React.cloneElement(pickerSelect, {
-        ...pickerSelect.props,
+      const item = pickerSelect as React.ReactElement<TimePickerSelectProps>;
+      if (item.type !== TimePickerSelect) {
+        return pickerSelect;
+      }
+
+      return React.cloneElement(item, {
+        ...item.props,
```

Comment:
I don't think we can do this without a breaking change. Right now it is not a requirement that a child be a `TimePickerSelect`. Consumers can use their own version of a TimePickerSelect-like component that may differ in signature from our `TimePickerSelect`.

Reply:
I see... then the only way I think is to make item `any` to let TS happy about the props we used in the second argument of `React.cloneElement`. I've tried to do a global search on the usage of `React.cloneElement` and see if there's existing solutions for this but it seems there's no existing ones already migrated to TS. Any thoughts?

Reply:
Used `as any` instead of type assertion in 1309904711bf91de11e7b3d1a0a6e42a2baac2b6